docs/guide/database-insert-update.md

```diff
@@ -0,0 +1,159 @@
+
+We then request the *full* database document. See the [database querying guide](/guide/database-queries.md) for more on this.
+
+```javascript
+dogDocument.receivedInterest++;
+
+const updateResult = await self.update(req, dogDocument);
+
+return updateResult;
+```
+
+We then update the document property that tracks how many times someone expressed interested in the dog. Then we use `self.update` to replace the previous document state with our update (with `await` as it is asynchronous). We finally return the result, which will be an object with update information from MongoDb.
+
+::: note
+**What if we are updating a document from a different doc type module?**
+
+In the example above, `self` refers to the `dog` piece type module since that is where the method is registered. If we wanted to run the `update` method from a separate module we would replace `self` with a specific reference to the `dog` module: **`self.apos.modules.dog`**.
+:::
+
+### Updating page documents
+
+Updating pages works the same way as pieces with the same arguments to the `update` method. There are a few things to keep in mind when working on pages, however.
+
+**We often call the page `update` method from `self.apos.page`.** `self.update` does work from individual page type modules and sometimes that is the right way to call it. But since pages can change their `type` property (unlike pieces) they share a single `update` method from the `@apostrophecms/page` module. `self.update` on individual page type modules is simply a wrapper that method.
+
+**There is a better way to move pages within the [page tree](/guide/pages.md#connecting-pages-with-page-tree-navigation).** The `@apostrophecms/page` module has a dedicated `move()` method for that purpose. Documentation of that method is upcoming. In the meantime, refer to the [REST API information about page tree placement](/reference/api/pages.md#post-api-v1-apostrophecms-page).
+
+## Inserting a new piece
+
+Inserting a new piece works very similarly to updating an existing one. The `self.insert` method takes the same arguments: `self.insert(req, piece, options)`. The big difference, of course, is that there is no existing document to find, update, and resubmit.
+
+Instead, use the `self.newInstance()` method to get a fresh document object of the proper document type. That method uses the doc type's [field schema](/guide/content-schema.md) to generate the essential document properties with any default values. We can then add any initial data to that essentially blank document object.
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    extend: '@apostrophecms/piece-type',
+    // ...
+    methods (self) {
+      async addNewDog(initialInfo) {
+        // Generate a req object using the contributor role
+        const req = apos.task.getReq({ role: 'contributor' });
+        // Generate a blank dog data object.
+        let newDog = self.newInstance();
```

Comment:
This example works better than the update example because it always makes sense to call `self.insert` (or page.insert) when creating a new thing.